### ISSUE TITLE
Building error. Unable to finish the installation.

### ISSUE THREAD

**Issue by HakaishinShwet** [wazuh/wazuh#25462]
![image](https://github.com/user-attachments/assets/f288cb4c-470c-4378-92b7-a0f81368a165)

Following is the error i am getting while trying to build package from this guide for my arch linux agent -

`https://documentation.wazuh.com/current/deployment-options/wazuh-from-sources/wazuh-agent/index.html`

```
client-agent/receiver.c: In function ‘receive_msg’:
client-agent/receiver.c:280:81: error: passing argument 4 of ‘UnmergeFiles’ from incompatible pointer type [-Wincompatible-pointer-types]
  280 |                                 if (!UnmergeFiles(file, SHAREDCFG_DIR, OS_TEXT, &ignore_list)) {
      |                                                                                 ^~~~~~~~~~~~
      |                                                                                 |
      |                                                                                 char ***
In file included from ./headers/shared.h:256,
                 from client-agent/receiver.c:11:
./headers/file_op.h:233:85: note: expected ‘const char ***’ but argument is of type ‘char ***’
  233 | int UnmergeFiles(const char *finalpath, const char *optdir, int mode, const char ***unmerged_files) __attribute__((nonnull(1)));
      |                                                                       ~~~~~~~~~~~~~~^~~~~~~~~~~~~~
client-agent/receiver.c:287:72: error: passing argument 2 of ‘cldir_ex_ignore’ from incompatible pointer type [-Wincompatible-pointer-types]
  287 |                                     if (cldir_ex_ignore(SHAREDCFG_DIR, ignore_list)) {
      |                                                                        ^~~~~~~~~~~
      |                                                                        |
      |                                                                        char **
./headers/file_op.h:358:54: note: expected ‘const char **’ but argument is of type ‘char **’
  358 | int cldir_ex_ignore(const char * name, const char ** ignore);
      |                                        ~~~~~~~~~~~~~~^~~~~~

Please guide what am i doing wrong so that i can build and install successfully for arch linux
```

**Comment by HakaishinShwet** [comment]
please guide me to fix this error for my arch linux

**Comment by MiguelazoDS** [comment]
Hi @HakaishinShwet, 

Since GCC 14 (The version that Arch Linux uses), some warnings now are treated as errors.

https://gcc.gnu.org/gcc-14/porting_to.html#warnings-as-errors

We already found this and the fix is intended to be released in version 4.9.0 

In the meantime, you can apply the fix in your code before compiling it. 

https://github.com/wazuh/wazuh/pull/23906

Regards!

**Comment by HakaishinShwet** [comment]
@MiguelazoDS  Thankyou very much , Finally was able to successfully compile after manually applying those commits :-))
